src/config/midi.rs

```diff
@@ -1,19 +1,69 @@
-use crate::Result;
-use serde::Deserialize;
-use wmidi::Channel;
+use serde::{Deserialize, Deserializer};
+use std::{convert::TryInto, ops::Deref};
+use wmidi::U7;
 
 #[derive(Debug, Deserialize)]
 pub struct Midi {
-    pub port: String,
-    channel: u8,
+    pub port: Option<String>,
 }
 
 impl Midi {
-    pub fn default_channel() -> Channel {
-        Channel::Ch1
+    pub fn default() -> Self {
+        Self { port: None }
     }
+}
+
+#[derive(Copy, Clone, Debug, Deserialize)]
+pub struct MidiSlider {
+    pub channel: Channel,
+    pub control_change: ControlChange,
+}
+
+#[derive(Copy, Clone, Debug, Deserialize)]
+#[serde(transparent)]
+pub struct Channel {
+    #[serde(deserialize_with = "deserialize_channel")]
+    value: wmidi::Channel,
+}
+
+impl Deref for Channel {
+    type Target = wmidi::Channel;
+
+    fn deref(&self) -> &Self::Target {
+        &self.value
+    }
+}
```

Comment:
Implementing `Deref` like this means that you can use our `Channel` everywhere that accepts a `wmidi::Channel`, so the overhead of having this wrapper type is not very high. (The `*` up above that I said I'd explain later was to manually invoke this `deref` because it couldn't be inferred for some reason in that case.)